dnf/base.py

```diff
@@ -2289,7 +2290,8 @@ def _verification_of_packages(pkg_list, logger_msg):
 
         try:
             msg = _('Package "{}" from local repository "{}" has incorrect checksum')
-            if not _verification_of_packages(local_repository_pkgs, msg.format(pkg, pkg.reponame)):
+            if pkg is not None and not _verification_of_packages(local_repository_pkgs,
```

Comment:
Correct line would be:
```
if not _verification_of_packages(local_repository_pkgs, msg):
```

Reply:
I try to little bit improve the code but I didn't test it. Please can you take a look and if you will like it, just use it.
```
diff --git a/dnf/base.py b/dnf/base.py
index 970f3003..515e2b9c 100644
--- a/dnf/base.py
+++ b/dnf/base.py
@@ -2285,11 +2285,18 @@ class Base(object):
         :return: list of remote pkgs
         """
         def _verification_of_packages(pkg_list, logger_msg):
+            all_packages_verified = True
             for pkg in pkg_list:
-                if not pkg.verifyLocalPkg():
-                    logger.critical(logger_msg)
-                    return False
-            return True
+                pkg_successfully_verified = False
+                try:
+                    pkg_successfully_verified = pkg.verifyLocalPkg()
+                except Exception as e:
+                    logger.critical(str(e))
+                if not pkg_successfully_verified:
+                    logger.critical(logger_msg.format(pkg, pkg.reponame))
+                    all_packages_verified = False
+
+            return all_packages_verified
 
         remote_pkgs = []
         local_repository_pkgs = []
@@ -2299,17 +2306,9 @@ class Base(object):
                     local_repository_pkgs.append(pkg)
             else:
                 remote_pkgs.append(pkg)
-        error = False
-
-        try:
-            msg = _('Package "{}" from local repository "{}" has incorrect checksum')
-            if not _verification_of_packages(local_repository_pkgs, msg.format(pkg, pkg.reponame)):
-                error = True
-        except Exception as e:
-            logger.critical(str(e))
-            error = True
 
-        if error:
+        msg = _('Package "{}" from local repository "{}" has incorrect checksum')
+        if not _verification_of_packages(local_repository_pkgs, msg):
             raise dnf.exceptions.Error(
                 _("Some packages from local repository have incorrect checksum"))
```